src/widget.rs

```diff
@@ -2,14 +2,20 @@
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
 
+#[cfg(any(feature = "v3_8", feature = "dox"))]
```

Comment:
We should maybe write a macro for things like this. It's getting a bit ugly when reading the code...